frontend/src/pages/auth/components/facebook-login-button/facebook-login-button.tsx

```diff
@@ -17,39 +11,59 @@ type FacebookLoginButtonProperties = {
   onLogin: (response: UserSignInWithFacebookResponseDto) => void;
 };
 
+type ProfileResponse = {
+  email: string;
+  id: string;
+};
+
+const FACEBOOK_ERROR_MESSAGE = 'Facebook sign in failed';
+
 const FacebookLoginButton: React.FC<FacebookLoginButtonProperties> = ({
   onLogin,
 }) => {
-  const responseFacebook = useCallback(
-    (response: ReactFacebookLoginInfo | ReactFacebookFailureResponse): void => {
-      if ('accessToken' in response) {
-        const { accessToken, email, name } = response;
-        onLogin({
-          accessToken,
-          email,
-          name,
-        });
-      } else {
-        notification.error('Facebook sign in failed');
+  const { init, isLoading } = useFacebook();
+
+  const handleFacebookLogin = useCallback(async (): Promise<void> => {
+    try {
+      const api = await init();
+
+      if (!api) {
+        throw new Error(FACEBOOK_ERROR_MESSAGE);
       }
-    },
-    [onLogin],
-  );
+
+      const response = await api.login({ scope: 'email' });
+
+      if (response.status !== LoginStatus.CONNECTED) {
+        throw new Error(FACEBOOK_ERROR_MESSAGE);
+      }
+
+      const profile = (await api.getProfile({
+        fields: ['email'],
+      })) as ProfileResponse | undefined;
+
+      if (!profile?.email) {
+        throw new Error(FACEBOOK_ERROR_MESSAGE);
+      }
+
+      onLogin({
+        accessToken: response.authResponse.accessToken,
+        email: profile.email,
+      });
+    } catch (error) {
+      error instanceof Error && notification.error(error.message);
+    }
+  }, [onLogin, init]);
```

Comment:
What if we put it in a separate hook?
But this is just an observation, there are no specific reasons for this, it just looks like this code can be moved from here.

Reply:
That's a specific method needed for specific button, so I'm not sure that we need to move it out